Source/santad/SNTExecutionController.mm

```diff
@@ -375,7 +375,7 @@ - (void)validateExecEvent:(const Message &)esMsg postAction:(bool (^)(SNTAction)
         // message to santad to perform the upload logic for bundles.
         // See syncBundleEvent:relatedEvents: for more info.
         se.needsBundleHash = YES;
-      } else if (config.syncBaseURL) {
+      } else if (config.syncBaseURL && (!config.enableStandaloneMode || cd.silentBlock)) {
```

Comment:
I think there might be an additional case here we didn't handle last time we looked at this. On line 413 below, we only setup the custom reply block if there isn't an explicit block rule (`se.decision == SNTEventStateBlockUnknown`). 

If there was an explicit block rule, (e.g. the decision was `SNTEventStateBlockBinary` or something) and `enableStandaloneMode` is set, then it seems the event would never be added to the sync queue. It wouldn't be added here because if check would fail. Then down below the custom `reply`block wouldn't be set either, resulting in no event being uploaded.

Reply:
Yeah thought about that and I think we need to figure out the right thing here or if I need to subclass the `SNTExecutionController` and just override it's behavior.

There's a few options here if we don't subclass:

1. Make the immediate upload happen if the decision is not `SNTEventStateBlockUnknown` and then add logging to the default replyBlock if `enableStandalone` mode and a `syncBaseURL` are both set for when a user dismisses the block dialog or fails authentication.

This is pretty easy and probably requires a helper function for the conditional as it's a little hairy and should be tested.

2. In standalone mode we can just defer all event uploads to the sync server until after the user has made the decision. 

This has the disadvantage of delaying when the sync service can display the event for approvals  / log or trigger bundle service interactions.

3. We can upload 2 events to the server, the first one if a `syncBaseURL` is configured (Santa's default behavior) and a second one if the user authenticates and approves the block. 

Option 3 would put an added burden on the sync service to determine what happened, but shouldn't be that crazy as the event identifiers and timestamps would all match.

Reply:
We've decided to go with option 4 a new event type to notify the sync service of local rule creation. 

This will come in a future PR.